docs/GettingStarted: document linker-related cmake options

nhaehnle opened the pull request:
Both LLVM_LINK_LLVM_DYLIB and LLVM_PARALLEL_LINK_JOBS help with some common gotchas. It seems worth documenting them here explicitly.

nhaehnle committed:
docs/GettingStarted: document linker-related cmake options

Both LLVM_LINK_LLVM_DYLIB and LLVM_PARALLEL_LINK_JOBS help with some
common gotchas. It seems worth documenting them here explicitly.

MaskRay commented:
I agree that `LLVM_PARALLEL_LINK_JOBS` is a frequently used option, but the option documentation duplication makes me nervous (status quo). I wonder whether this section should just list option names and refer readers to `CMake.html`.

nhaehnle committed:
Keep the list of variable names but refer to CMake.rst for details

I took the liberty of moving LLVM_ENABLE_DOXYGEN to keep
thematically-related variables closer together.

MaskRay commented on llvm/docs/GettingStarted.rst:
```diff
@@ -540,75 +540,23 @@ Variables are passed to ``cmake`` on the command line using the format
-|                         | the LLVM build. This can dramatically speed up     |
-|                         | debug builds.                                      |
-+-------------------------+----------------------------------------------------+
+* CMAKE_C_COMPILER
```
Add double backticks to quote each variable

nhaehnle committed:
Add double backticks
